platform-sdk/docs/components/componentFramework.md

```diff
@@ -0,0 +1,218 @@
+# Component Framework
+
+## Goals
+
+The goal of this framework is to provide a structure for defining software components in which:
+
+1. Components can be defined in isolation, each with their own unit of business logic.
+2. Components can be connected (or wired) together with other components to form a data pipeline.
+3. Each component allows its threading model to be changed individually without modifying the business
+   logic.
+
+These goals allow software to be well organized, flexible, and easily tuned for performance.
+
+## Overview
+
+At the most basic level, a component in this framework is composed of a task scheduler, business logic, and wires. Each
+component has a unit of business logic that performs some action on a piece of data. The component receives the data on
+an input wire, performs the action, and optionally produces data on an output wire. The components are connected, or
+"soldered", together by joining input and output wires of the same data type to form a data pipeline.
+
+The concurrency of the component is determined by the task scheduler type. Each component has a task scheduler which
+combines the data received on the input wire with the business logic operation to form a task. This task is then
+scheduled for execution according to the task scheduler type and implementation. Task schedulers are described in more
+detail [below](#TaskSchedulers).
+
+Here is a picture of a simple component with one input wire of type `Integer`, and one output wire of type `String`:
+
+![SimpleComponent](step4.drawio.svg)
+
+### Business Logic
+
+Business logic is defined in one or more classes that work together and accept data, perform an
+operation, and produce optional output. This framework allows business logic to be defined completely independently
```

Comment:
It would be helpful to look at `BindableInputWire.bind()`. The business logic method is bound to the input wire in a lambda function. The lambda function checks the return of the business logic function and only forwards data to the task scheduler to be put on the output wire if the return is non-null.

The details of the implementation are slightly different than the diagram. I hoped to capture the conceptual flow to give readers enough context to aid them in digging into the details further via the code. Do you think it is worth explaining the nuance of how this works in the document?

Reply:
Thanks! It makes sense.
I wanted to know if there was some other filtering responsibility to be added between components or if the next component in the chain needed to handle the optionality of the previous one... That would be worth noting, but
as the diagram is, it captures how it works...